doc/compiler/tril/TrilLanguageReference.md

```diff
@@ -0,0 +1,228 @@
+     case for certain opcodes. 
+   * `@exit` falls through to the exit block of the CFG. 
+
+#### Example
+
+```
+(method ...
+   (block fallthrough="sub" name="start"
+      (ificmpge target="add"                 
+        ...) )
+   (block name="sub"
+      (ireturn
+         ...) )
+   (block name="add"
+      (ireturn                               
+        ... ) )
+```
+   
+### Commoning 
+
+Nodes can be given identifiers to allow creating commoned subtrees. To common a
+reference, on first compuation annotate the node with the `id="<some name>"` property. Then,
+where commoning is desired, replace the node with `(@id "<some name>")` 
+
+```
+(imul id="multiplyResult"
+    ...) 
+(iadd
+  (@id "multiplyResult")
+  (@id "multiplyResult")
+```
+
+### Vectors 
+
+Vector support in Tril is very preliminary, but present. One of the most
+important pieces is the ability to annotate indirect vector loads and stores
+with the data type of the load. 
+
+This is required because the vector support in Testarossa infers the vector
+types from certain roots, as opposed to expressing the data types in the
+opcodes directly like other opcodes in Testarossa. 
+
+    (vloadi type=VectorDouble (aload parm=0))
+    
+####Properties
+
+* `type` _Mandatory_ The data type being loaded or stored. 
+
+
+
+### Calls
+
+Call opcodes are supported with in Tril by providing extra annotation of the
+argument types and the function pointer to be invoked.
+
+    (icall address=0x1234 args=[Int32] ...) 
+
+####Properties
+
+* `address` _Mandatory_ The address of the function to be invoked by the call opcode
+* `args` _Optional_ The list of argument types passed to the call. 
+
+
+# Examples 
+
+## Incordec 
+
+Checked in as `fvtest/tril/examples/incordec/incordec.tril`
+
+```
+; This simple method takes as argument a pointer to a 32-bit integer. If the
+; integer pointed to by the argument has a negative value, then the value
+; decremented by 1 is returned. Otherwise (value is positive), the value
+; incremented by 1 is returned.
+;
+; An equivalent C implementation:
+;
+; int incordec(int* parm0) {
+;    if (*value < 0) return *parm - 1;
+;    else return *parm + 1;
+; }
+
+(method name="incordec" return="Int32" args=["Address"]
+   (block fallthrough="sub" name="start"     ; start:
+      (ificmpge target="add"                 ; if (*parm0 >= 0) goto add;
+         (iloadi offset=0 (aload parm=0) )
+         (iconst 0) ) )
+   (block name="sub"                         ; sub:
+      (ireturn                               ; return *parm0 - 1;
+         (isub
+            (iloadi offset=0 (aload parm=0) )
+            (d2i (dconst 1.0) ) ) ) )
+   (block name="add"                         ; add:
+      (istore temp="x" (iconst 1))           ; int x = 1;
+      (ireturn                               ; return *parm0 + x;
+         (iadd
+            (iloadi offset=0 (aload parm=0) )
+            (iload temp="x") ) ) ) )
+```
```

Comment:
While functionally equivalent to the C snippet above, I think its worth pointing out that you exercised your creative license in order to demonstrate some additional features of Tril in order to achieve that equivalent functionality (like the conversion of the dconst and the store/load from a temporary).

Reply:
Done.